osde2e-wrapper.py

```diff
@@ -172,20 +157,39 @@ def _build_cluster(osde2e_cmnd,account_config,my_path,es,index,my_uuid,my_inc,ti
     yaml.dump(account_config,open(cluster_path + "/cluster_account.yaml",'w'))
     cluster_env = os.environ.copy()
     cluster_env["REPORT_DIR"] = cluster_path
-    if account_config['ocm']['expiration']:
+    if "expiration" in account_config['ocm'].keys():
         cluster_env["CLUSTER_EXPIRY_IN_MINUTES"] = str(account_config['ocm']['expiration'])
     logging.info('Attempting cluster installation')
     logging.info('Output directory set to %s' % cluster_path)
     cluster_cmd = [osde2e_cmnd, "test","--custom-config", "cluster_account.yaml"]
     if not dry_run:
         process = subprocess.Popen(cluster_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=cluster_env, cwd=cluster_path)
         stdout,stderr = process.communicate()
+        cluster_end_time = time.strftime("%Y-%m-%dT%H:%M:%S")
         if process.returncode != 0:
             logging.error('Failed to build cluster number %d' % my_inc)
             logging.error(stderr.strip().decode("utf-8"))
             success = False
+        logging.info('Attempting to load metadata json')
+        try:
+            metadata = json.load(open(cluster_path + "/metadata.json"))
+        except Exception as err:
+            logging.error(err)
+            logging.error('Failed to load metadata.json file located %s' % cluster_path)
+        metadata["cluster_start_time"] = cluster_start_time
+        metadata["cluster_end_time"] = cluster_end_time
+        metadata["install_successful"] = success
+        metadata["uuid"] = my_uuid
+        metadata["install_counter"] = my_inc
+        try:
+            with open(cluster_path + "/metadata.json", "w") as metadata_file:
+                json.dump(metadata, metadata_file)
+        except Exception as err:
+            logging.error(err)
+            logging.error('Failed to write metadata.json file located %s' % cluster_path)
         if es is not None:
-            _index_result(es,my_uuid,index,cluster_path + "/metadata.json",cluster_start_time,success,timestamp)
+            metadata["timestamp"] = time.strftime("%Y-%m-%dT%H:%M:%S")
```

Comment:
Previously timestamp was the same for all clusters requested per invocation. This changes that data to be different per cluster and almost identical to the cluster_end_time. I would vote to keep it with the old functionality for ease of searching in ES but would love to know your thoughts.

Reply:
timestamp identifies the time where data is uploaded to ES. Thats why it is being calculated only if it will be uploaded to ES after the installation and not saved to the metadata.json file 

If it is uploaded during the installation I agree that it will be close to the cluster_end_time, but we need to calculate at the time it is being to be uploaded, because if we execute the script with the new `--es-index-only` parameter, it wont be consistent with the time that data is uploaded to ES